wire-ios-sync-engine/Source/Synchronization/Strategies/CallingRequestStrategy.swift

```diff
@@ -332,22 +336,36 @@ extension CallingRequestStrategy: WireCallCenterTransport {
 
             self.zmLog.debug("schedule calling message")
 
-            let genericMessage = GenericMessage(content: Calling(content: dataString))
             let recipients = targets.map { self.recipients(for: $0, in: self.managedObjectContext) } ?? .conversationParticipants
 
-            let message = GenericMessageEntity(
-                conversation: conversation,
-                message: genericMessage,
-                targetRecipients: recipients,
-                completionHandler: nil
-            )
+            let message: GenericMessageEntity
+
+            if overMLSSelfConversation, conversation.messageProtocol == .mls {
+                guard let selfConversation = ZMConversation.fetchSelfMLSConversation(in: self.managedObjectContext) else {
+                    WireLogger.mls.error("missing self conversation for sending message to own clients")
+                    completionHandler(500)
+                    return
+                }
+
+                message = GenericMessageEntity(conversation: selfConversation,
+                                               message: genericMessage,
+                                               targetRecipients: recipients,
+                                               completionHandler: nil)
+            } else {
+                message = GenericMessageEntity(
+                    conversation: conversation,
+                    message: genericMessage,
+                    targetRecipients: recipients,
+                    completionHandler: nil
+                )
+            }
 
             switch (conversation.messageProtocol, recipients) {
             case (.proteus, _), (.mls, .conversationParticipants):
                 message.send(with: self.messageSync, completion: completionHandler)
 
             case (.mls, _):
-                if message.isConferenceKey {
+                if overMLSSelfConversation {
```

Comment:
I don't think this is right... I think we need to send the conference key to all the members of the conference. Before subconversations, we needed to send it to the entire parent conversation, but now I think we have (not on this branch though) subconversations we can target the conference members.

We'll need to revise this workaround on the other branch though.

Reply:
ok so I'll keep it as it is and add a TODO when subconversations are available